src/commands/circleci/envvar/update.ts

```diff
@@ -0,0 +1,229 @@
+  public static readonly description = messages.getMessage('envvar.update.description');
+  public static readonly examples = messages.getMessage('envvar.update.examples').split(EOL);
+
+  public static readonly flagsConfig: FlagsConfig = {
+    slug: flags.string({
+      description: messages.getMessage('envvar.update.flags.slug'),
+      char: 's',
+      multiple: true,
+    }),
+    envvar: flags.string({
+      description: messages.getMessage('envvar.update.flags.envvar'),
+      char: 'e',
+      required: true,
+      multiple: true,
+    }),
+    dryrun: flags.boolean({
+      description: 's',
+    }),
+  };
+
+  private envvarValues: { [index: string]: string } = {};
+
+  public async run(): Promise<CircelCIEnvvarUpdateStatus> {
+    if (this.flags.dryrun) {
+      this.ux.log(
+        yellow('Dryrun mode set. All validation will occur but the environment variables will NOT be updated.') + EOL
+      );
+    }
+
+    const slugs: string[] = await this.resolveSlugs();
+
+    await this.resolveEnvvarValues();
+
+    const status: CircelCIEnvvarUpdateStatus = {};
+    for (const slug of slugs) {
+      const batch = await this.updateEnvvars(slug);
+      status[slug] = batch;
+      this.printStatus(slug, batch);
+    }
+
+    return status;
+  }
+
+  private get headers(): { [index: string]: string } {
+    const token = env.getString('CIRCLE_CI_TOKEN');
+    if (!token) {
+      throw new SfdxError('The environment variable "CIRCLE_CI_TOKEN" is required.');
+    }
+    return {
+      'Circle-Token': token,
+    };
+  }
+
+  private async resolveEnvvarValues(): Promise<void> {
+    for (const envvarName of this.getFlagAsArray('envvar')) {
+      const envvarValue = env.getString(envvarName);
+      if (envvarValue) {
+        this.envvarValues[envvarName] = envvarValue;
+      } else if (!(await this.isPipedIn())) {
+        this.envvarValues[envvarName] = await this.ux.prompt(envvarName, { type: 'mask' });
+      } else {
+        throw new SfdxError(`missing envvar value for ${envvarName}`);
+      }
+    }
+  }
+
+  private async updateEnvvars(slug: string): Promise<string | EnvvarUpdateStatus[]> {
+    const envvarNames: string[] = this.getFlagAsArray('envvar');
+    let response;
+
+    try {
+      response = await got.get(`${URL_BASE}/${slug}/envvar`, { headers: this.headers });
+    } catch (error) {
+      return `${error.message as string}. Skipping...`;
+    }
+    const body = JSON.parse(response.body) as { items: [{ name: string }] };
+    const existingEnvvars = body.items;
+
+    const notFoundEnvvars = envvarNames.filter(
+      (envvarName) => !existingEnvvars.find((existingEnvvar) => existingEnvvar.name === envvarName)
+    );
+
+    if (notFoundEnvvars.length > 0) {
+      const envvarList = envvarNames.join(', ');
+      const notFoundList = notFoundEnvvars.join(', ');
+      return `Envvars [${envvarList}] not set. ALL specified envvars [${notFoundList}] must be set on the slug. Skipping...`;
```

Comment:
Just curious, is this a limitation of the circle api? Because it might be nice at some point to be able to setup the env vars for a new plugin programmatically

Reply:
It was because I didn't want someone to accidentally update AWS keys on repo that didn't need them. I could have left this out and the `got.delete` would have failed, but figured I would be proactive and prevent all if one was missing.

I can see it being nice to setup new repos though or set a new env var across a bunch of repos...

Options (in order of my preference):
1. Add a `--allow-new` flag.
1. Have both the `set` and `update` command.
1. Change this to `circleci:envvar:set` with a `--replace` flag.
1. Leave it as is for now.
1. Remove this and put a try/catch around the `delete` so it will update or set.